### PR TITLE
Add all serenata supported initial config options

### PR DESCRIPTION
**Pull request by Sasanidas**
Add support for all the Serenata [config](https://gitlab.com/Serenata/Serenata/-/wikis/Setting%20Up%20Your%20Project) options.

I had to do some workaround in the uris parameter, because all has to be filled with data, setting up the root workspace by hand for every project seems a little inconvenience, so I decided to set a default value base on the workspace root location given by `lsp-workspace-root`.

Maybe it need more documentation or there is a better way to do it, as always, I'm open to suggestions :+1:

### CODE REVIEW

File: lsp-serenata.el [hunk 1]
```diff
@@ -39,21 +40,98 @@ It can be downloaded from https://gitlab.com/Serenata/Serenata/-/releases."
...
   :major-modes '(php-mode)
   :priority -2
-  :server-id 'serenata
+  :notification-handlers (ht ("serenata/didProgressIndexing"
+			      (lambda (_server data)
+				(lsp--info "%s" (lsp:serenata-did-progress-indexing-info data)))))
+  :initialization-options #'lsp-serenata-init-options
   :initialized-fn (lambda (workspace)
+		    (when (equal (length lsp-serenata-uris) 0)
+		      (let* ((lsp-root (concat "file://" (lsp-workspace-root))))
```
Comment (yyoncho) [inline lsp-serenata.el:129]: use `lsp--path-to-uri`

File: lsp-serenata.el [hunk 2]
```diff
@@ -39,21 +40,98 @@ It can be downloaded from https://gitlab.com/Serenata/Serenata/-/releases."
...
+included by default as it is often used to contain test code that
+is not directly part of the code.  Note that for existing
+projects, removing extensions will not not automatically prune
+files having them from the index if they are already present.
+Adding new ones will cause the files having them to be picked up
+on the next project initialization."
+  :group 'lsp-serenata
+  :type 'lsp-string-vector)
+
+(defcustom lsp-serenata-index-database-uri (concat "file://"(getenv "HOME") "/.emacs.d/index.sqlite")
```
Comment (yyoncho) [inline lsp-serenata.el:82]: lsp--path-to-uri + f-join

**Comment by Sasanidas** [comment]
> Looks good, can you rebase on master?

Ok, done :+1:

File: lsp-serenata.el [hunk 3]
```diff
@@ -39,21 +40,98 @@ It can be downloaded from https://gitlab.com/Serenata/Serenata/-/releases."
...
+included by default as it is often used to contain test code that
+is not directly part of the code.  Note that for existing
+projects, removing extensions will not not automatically prune
+files having them from the index if they are already present.
+Adding new ones will cause the files having them to be picked up
+on the next project initialization."
+  :group 'lsp-serenata
+  :type 'lsp-string-vector)
+
+(defcustom lsp-serenata-index-database-uri (lsp--path-to-uri (f-join  "~/.emacs.d/index.sqlite"))
```
Comment (yyoncho) [inline lsp-serenata.el:82]: dont you need home directory here?

Comment (Sasanidas) [reply lsp-serenata.el:82]: Yes, it still works with GNU/Linux, but it may break in Windows/Mac I don't know.
I'll add the user-emacs-directory variable

**Comment by yyoncho** [comment]
There are issues with the rebase?

> Conflicting files
> lsp-serenata.el

**Comment by Sasanidas** [comment]
> There are issues with the rebase?
> 
> > Conflicting files
> > lsp-serenata.el

Ups! I didn't saw the conflicts, sorry

**Comment by yyoncho** [comment]
@Sasanidas thank you!

**Pull request merged**